engine/lib/ast.ml

```diff
@@ -338,8 +338,11 @@ functor
           witness : F.nontrivial_lhs;
         }
 
+    and guard = { guard : guard'; span : span }
```

Comment:
Can you add:
 - a doc comment that explains what is a `guard` (with an example)
 - a comment on `guard'` explaining why there is only one variant currently, and a link to an issue that explains the roadmap for the work left on guards (that is, simple lets and let-chains), with link to Rust documentation on that